test/user.test.js

```diff
@@ -1921,6 +1949,75 @@ describe('User', function() {
       });
     });
 
+    describe('User.verify(id)', function() {
+      beforeEach(()=> {
+        User.getVerifyOptions = function() {
+          const base = User.base.getVerifyOptions();
+          return Object.assign({}, base, {
+            from: 'noreply@email.com',
+          });
+        };
+      });
+
+      it('triggers user identity verification', function() {
+        return User.create({email: 'bar@bat.com', password: 'bar'})
+        .then(user => {
+          return User.verify(user.pk);
+        })
+        .then(res => {
+          // No need to redo all the tests already done in the User.prototype.verify()
+          // prototype method: just doing a partial check to confirm that the
+          // prototype method was called successfully.
+          expect(res.email.envelope).to.eql({from: 'noreply@email.com', to: ['bar@bat.com']});
+        });
+      });
+
+      it('throws when providing an invalid user id', function() {
+        return User.create({email: 'bar@bat.com', password: 'bar'})
+        .then(user => {
+          return User.verify('invalidUserId');
+        })
+        .then(
+          function onSuccess() {
+            throw new Error('User.verify() should have failed');
+          },
+          function onError(err) {
+            err = Object.assign({}, err);
+            expect(err).to.eql({
+              code: 'USER_NOT_FOUND',
+              statusCode: 422,
+            });
+          }
+        );
+      });
+
+      it('is called over REST method /User/verify', function() {
+        return request(app)
+          .post('/test-users')
+          .send({email: 'bar@bat.com', password: 'bar'})
+          .expect('Content-Type', /json/)
+          .expect(200)
```

Comment:
Since this test is checking the REST API of the verify function only, I'd prefer to create the user using regular JS API - it makes the test easier to read to me.

```js
return User.create(...).then(u => {
  return request(app)
    // ...
});
```

Thoughts?